tools/forcerelay-test/src/lib.rs

```diff
@@ -0,0 +1,159 @@
+#[cfg(test)]
+mod tests {
+    use ckb_sdk::constants::TYPE_ID_CODE_HASH;
+    use ckb_sdk::{
+        rpc::ckb_light_client::{Order, ScriptType, SearchKey},
+        IndexerRpcClient,
+    };
+    use ckb_types::packed;
+    use ckb_types::{core::ScriptHashType, prelude::*};
+    use std::process::Command;
+    use std::thread;
+    use std::time;
+    use std::time::Duration;
+
+    use eth_light_client_in_ckb_verification::types::packed::{ClientReader, Uint64};
+    #[ignore]
+    #[test]
+    fn integration_test() {
+        let ckb_path = "ckb-dev";
+        let mut working_dir = std::env::current_dir().unwrap();
+        working_dir.push(ckb_path);
+
+        let _ = std::fs::remove_dir_all(ckb_path);
+        std::fs::create_dir(ckb_path).unwrap();
+
+        let (mut ckb_run, mut ckb_miner) = {
+            Command::new("ckb")
+                .arg("init")
+                .arg("--chain")
+                .arg("dev")
+                .current_dir(&working_dir)
+                .spawn()
+                .unwrap();
+
+            thread::sleep(Duration::from_secs(5));
+
+            std::fs::copy("ckb/ckb.toml", format!("{}/ckb.toml", ckb_path)).unwrap();
+            std::fs::copy("ckb/dev.toml", format!("{}/specs/dev.toml", ckb_path)).unwrap();
+
+            let ckb_run = Command::new("ckb")
+                .arg("run")
+                .current_dir(&working_dir)
+                .spawn()
+                .unwrap();
+
+            thread::sleep(Duration::from_secs(1));
+
+            let ckb_miner = Command::new("ckb")
+                .arg("miner")
+                .current_dir(&working_dir)
+                .spawn()
+                .unwrap();
+
+            thread::sleep(Duration::from_secs(10));
+
+            let tx = std::fs::read_to_string("ckb/deploy-contracts.json").unwrap();
+            let rpc_content = format!(
+                r#"{{"id": 3, "jsonrpc": "2.0", "method": "send_transaction", "params": [{}]}}"#,
+                tx
+            );
+
+            // deploy the contracts
+            Command::new("curl")
+                .arg("-H")
+                .arg("content-type: application/json")
+                .arg("-d")
+                .arg(rpc_content)
+                .arg("http://localhost:8114")
+                .spawn()
+                .unwrap();
+
+            thread::sleep(Duration::from_secs(10));
+
+            (ckb_run, ckb_miner)
+        };
+
+        let mut running_eth_server = Command::new("python3")
+            .arg("eth_mock/server.py")
+            .spawn()
+            .unwrap();
+
+        // Waiting for the finality of the ckb contracts
```

Comment:
dont't need to wait for that so long, you can just wait for one block of native ckb mining.